cli/src/commands/split.rs

```diff
@@ -51,21 +51,61 @@ use crate::ui::Ui;
 /// Split a revision in two
 ///
 /// Starts a [diff editor] on the changes in the revision. Edit the right side
-/// of the diff until it has the content you want in the new revision. Once
-/// you close the editor, your edited content will replace the previous
-/// revision. The remaining changes will be put in a new revision on top.
+/// of the diff until it has the content you want in the first commit. Once you
+/// close the editor, your revision will be split into two commits.
 ///
 /// [diff editor]:
 ///     https://docs.jj-vcs.dev/latest/config/#editing-diffs
 ///
+/// By default, the selected changes stay in the original commit, and the
+/// remaining changes go into a new child commit:
+///
+/// ```text
+/// C                 C'
+/// |                 |
+/// B (split)   =>    B" (remaining)
+/// |                 |
+/// A                 B' (selected)
+///                   |
+///                   A
+/// ```
+///
+/// With `--parallel/-p`, the two parts become sibling commits instead of
+/// parent and child:
+///
+/// ```text
+///                   C'
+/// C                /  \
+/// |               B"   B' (selected)
+/// B (split)  =>    \  /
+/// |                 A
+/// A
+/// ```
+///
+/// With `-o`, `-A`, or `-B`, the selected changes are extracted into a new
+/// commit at the specified location, while the remaining changes stay in place:
+///
+/// ```text
+/// D                 D
+/// |                 |
+/// C                 C
+/// |                 |
+/// B (split)   =>    B' (remaining, stays here)
+/// |                 |
+/// A                 A
+///                   |
+///                   X (selected, inserted after A with -A A)
```

Comment:
nit: the `jj rebase` help uses commit identifiers starting at `J` instead of `A` in order to avoid confusion between `-A/-B` and commit identifiers

Reply:
I was going to suggest that.